Src/Metrics.Tests/Reporting/EventMetricsCleanerTest.cs

```diff
@@ -0,0 +1,212 @@
+﻿
+using System;
+using System.Collections.Generic;
+using FluentAssertions;
+using Metrics.Core;
+using Metrics.MetricData;
+using Metrics.Reporters.Cleaners;
+using Metrics.Utils;
+using Xunit;
+
+namespace Metrics.Tests.Reporting
+{
+    public class EventMetricsCleanerTest
+    {
+        [Fact]
+        public void StartsWithZeroReports()
+        {
+            EventMetricsCleaner.Clear();
+
+            EventMetricsCleaner.TotalReports.Should().Be(0);
+        }
+
+        [Fact]
+        public void DefaultIntervalIsGreaterThanZero()
+        {
+            EventMetricsCleaner.Clear();
+
+            EventMetricsCleaner.CurrentIntervalSeconds.Should().BeGreaterThan(0);
+        }
+
+        [Fact]
+        public void RegisterReport_IncrementsReportCount()
+        {
+            EventMetricsCleaner.Clear();
+
+            EventMetricsCleaner.RegisterReport(new TimeSpan(0));
+
+            EventMetricsCleaner.TotalReports.Should().Be(1);
+        }
+
+        [Fact]
+        public void RegisterReport_WithGreaterTimespan_UpdatesIntervalToBeGreaterThanReportInterval()
```

Comment:
Could test WithSmallerTimespam_DoesNotUpdateInterval

Reply:
Have added a unit test to test the correct logic.